### ISSUE TITLE
No wakeup on meshagent clien MacOS Sonoma

### ISSUE THREAD

**Issue by tootai** [Ylianst/MeshAgent#238]
Hi

when connected to a MacOS Sonoma -here M1 MBA- as soon as the Mac screen goes blank in energy save mode after x minutes of inactivity, it's impossible to wake up the agent. Computer is shown as connected -eg all tabs are apparent but connected button is grey- but wake up or sending any key combination does'nt have any effect.

On the Mac, by pressing any key it goes back to live as well as mesh connection after a while. We see

Connecting to: wss://mesh.domain.tld:443/agent.ashx
2024-04-04 05:14:46 PM: Control Channel Idle Timeout = 120 seconds
Connected.
Server verified meshcore... meshcore already running...
Mesh Server Connection Error [14]
Connecting to: wss://mesh.domain.tld:443/agent.ashx
2024-04-04 05:14:52 PM: Control Channel Idle Timeout = 120 seconds
Connected.
Server verified meshcore... meshcore already running...
Server verified meshcore... mes hcore already running...
Mesh Server Connection Error [15]

Daniel

**Comment by si458** [comment]
i believe ive fixed this recently in version 1.1.22, what version are you running?
i used `caffeinate` to wake up the mac screen to the login screen

**Comment by tootai** [comment]
1.1.22 as you

**Comment by si458** [comment]
sorry just realised you meant, you cant even connect to the mac, as the connect button is greyed out.
its oversally the mac is going to sleep and disconnecting the lan port (as such) so it cant connect to meshcentral
the `wake` feature in the actions, only sends wake on lan packets IF machines in the same group are in the same network as the machine you want to switch on

**Comment by si458** [comment]
do you have 'wake for network access' enabled?
i have macbook pro 2019 intel/macmini m1 both running sonoma here and they seem to stay connected with no issues,
also where are you seeing the logs?

**Comment by si458** [comment]
@tootai sorry i meant WHERE are those logs?
you generally only see those logs if you are running meshagent in a console mode, rather than it running in the background?
how did you install meshagent?

**Comment by tootai** [comment]
meshagent is normally installed and started on demand by the user using a script of mine in Applications/Utilities. The script is
#!/bin/bash
sudo pkill -9 meshagent
sudo /usr/local/mesh_services/meshagent/meshagent
sudo pkill -9 meshagent

and running in Terminal. User enter his password, all is good I have access,  finally CTRL/C to exit

**Comment by tootai** [comment]
forgot, logs are those from terminal

**Comment by si458** [comment]
@tootai ah yes this is why you are having issues.
because the app wont run when the laptop goes to sleep as the mac shuts down (in a sense).
as for why its staying connected but u cant connect, this is probably just a delay which happens even with windows computers.
meshcentral waits so long before it then says oh ove had no reply, this device is offline!
you could try installing the agent in the background using the mpkg or `sudo /usr/local/mesh_services/meshagent/meshagent -install`

**Comment by tootai** [comment]
Yes but in this case agent will start as soon as computer is started and will keep running, that's not my goal, for other OSs too. I use meshcentral to replace Anydesk/Teamviers/aso I don't want to have customers always connected !

**Comment by si458** [comment]
i dont think the is much you can do sadly, as i dont thitnk you can keep the meshagent running/connected when the mac sleeps because its running as a user process and not as a service/root user, even tho you use `sudo`
im sorry about this

**Comment by tootai** [comment]
Power settings are already setted to reactivated=always for network access. Will try if -install can be an option.

**Comment by tootai** [comment]
I ran -install which uninstalled/reinstalled meshagent. I gave all needed authorizations to meshagent, it change nothing: after 2 minutes MBA screen goes blank and connection is broken.

Interesting is, while connection is broken, MBA has an IP change (here I connect it to another Wireless Network) when I wake it up, meshcentral still show it disconnected but with old parameters in details like IP address. This brings 2 questions:
. how to telle meshcentral to forget about this connection and do a reset (only wake up is possible)
. why meshagent doesn't note the network modifications and send new datas

**Comment by tootai** [comment]
1st interrogation is useless, problem is 2nd one: if meshagent reconnect, Q1 will be solved.

**Comment by si458** [comment]
the Details tab only gets it information updated every 15 minutes, its not instant.
you can force update it by running `sysinfo` in the console tab

i have just noticed, that when my macbook pro goes to sleep, its also disconnecting from meshcentral (im running intel not arm), so i am mistaken!
however my macmini (arm) which is hardwired, never drops connection and always stays online,
im guessing this must just be a laptop thing, and i dont think the is anything we can do im afraid

**Comment by tootai** [comment]
OK. Another one: I installed meshagent with -install but I can't see it in Applications nor Utilities. Where should I look for? My goal is to try to create a keyboard shortcut to start/stop the service, something like services in Windows. Don't know why but my script isn't allow -grey marked- when I tried to create a shortcut.

**Comment by si458** [comment]
you wont see an application, because when you install it, its runs in the background 24/7 as a service.
macos doesnt have services as such, it uses launchctl

one way to stop the service is `sudo launchctl unload /Library/LaunchDaemons/meshagent.plist`
and to start it again you run `sudo launchctl load /Library/LaunchDaemons/meshagent.plist`

**Comment by tootai** [comment]
I modified my script like

#!/bin/sh
sudo launchctl unload /Library/LaunchDaemons/meshagent.plist
[ "$1" != "stop" ] && sudo launchctl load /Library/LaunchDaemons/meshagent.plist
exit 0

and in meshagent dir
ln -s /Applications/Utilities/myscript myscript

This way I'm more MacOS compatible. I close meshagent by going to Meshcentral <client> Terminal tab => connect and run ./myscript stop

It's a pity that we can't wake up a MacOS client but that's life ;)

Thanks for your support

**Comment by si458** [comment]
@tootai no worries!
In theory hardwired macs still work when they goto sleep, ots just wireless ones that don't, but I will have a look when I get chance as I have one of both to test hehe
